### PR TITLE
#25 remove revoked keys from key cache

### LINKED ISSUE #25: Allow revoking keys
We are currently never removing public keys from our `keyCache`, we are only adding keys:

https://github.com/zalando/planb-tokeninfo/blob/master/keys/openidprovider.go#L73

This prevents us currently from revoking keys in the provider.

### PR DESCRIPTION
**Pull request by hjacobs**
Fixes #25

### CODE REVIEW

**Comment by lmineiro** [comment]
Also tried to make golint happy :)

File: handlers/tokeninfo/errors.go [hunk 1]
```diff
@@ -6,22 +6,25 @@ import (
...
+	// ErrInvalidRequest should be used whenever the receiver failed to parse the request
+	ErrInvalidRequest = Error{"invalid_request", "Access Token not valid", http.StatusBadRequest}
+	// ErrInvalidToken should be used whenever the receiver failed to validate a JWT Token
+	ErrInvalidToken = Error{"invalid_token", "Access Token not valid", http.StatusUnauthorized}
 )
 
-// error response format for the token info endpoint
-func Error(w http.ResponseWriter, terr TokenInfoError) {
+// WriteError will write the Error e to the response writer, marshaled as JSON, and with the respective Status Code
+func WriteError(w http.ResponseWriter, e Error) {
```
Comment (aryszka) [inline handlers/tokeninfo/errors.go]: i would try to accept the generic error type, too, and fallback to some default status code and the err.Error() as the description. Or if really only the tokeninfo.Error is acceptable, then the Write method could have it as a receiver, like err.Write(w http.ResponseWriter). just nitpicking, though.

**Comment by aryszka** [comment]
lgtm

File: handlers/tokeninfo/errors.go (hunk 1 above)
Comment (lmineiro) [inline handlers/tokeninfo/errors.go]: Very good idea. Thanks

**Comment by hjacobs** [comment]
LGTM

**Pull request merged**